### PR TITLE
Migrate the Next.js template to Circuit UI v7

### PR DESCRIPTION
**Pull request by connor-baer**
## Approach and changes

- Upgraded all design system dependencies to their latest version
- Replaced Emotion.js with CSS Modules
- Reduced the total JavaScript size of the template homepage from 97.8 kb to 82.3 kb (gzipped), a 16% improvement
- Redesigned the template homepage:

| Before                                                                                                                                                      | After                                                                                                                                                            |
|-------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Screenshot 2023-07-15 at 17-56-43 Welcome to SumUp Next js | Screenshot 2023-07-21 at 12-57-02 Welcome to Circuit UI Next js |

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

File: packages/cna-template/template/components/Meta/Meta.tsx [hunk 1]
```diff
@@ -59,14 +54,6 @@ export const Meta: FC<PropsWithChildren<MetaProps>> = ({
...
       {updatedAt && <meta property="og:updated" content={updatedAt} />}
 
-      {/* Twitter */}
-      {twitter && <meta name="twitter:site" content={`@${twitter}`} />}
-      {image && <meta name="twitter:image:alt" content={image.alt} />}
-      <meta
-        name="twitter:card"
-        content={image ? 'summary_large_image' : 'summary'}
-      />
-
```
Comment (tareqlol) [inline packages/cna-template/template/components/Meta/Meta.tsx:69]: What is the reason behind this removal?

File: packages/cna-template/template/test-utils.tsx [hunk 2]
```diff
@@ -1,17 +1,4 @@
...
-import { light } from '@sumup/design-tokens';
-
-const WithProviders = ({ children }: { children: ReactNode }) => (
-  <ThemeProvider theme={light}>{children}</ThemeProvider>
-);
-
-export const render = (
-  component: Parameters<typeof renderTest>[0],
-  options: Parameters<typeof renderTest>[1] = {},
-) => renderTest(component, { wrapper: WithProviders, ...options });
```
Comment (tareqlol) [inline packages/cna-template/template/test-utils.tsx:14]: 🧹

File: packages/cna-template/template/components/Meta/Meta.tsx (hunk 1 above)
Comment (connor-baer) [reply packages/cna-template/template/components/Meta/Meta.tsx:69]: I refuse to support Twitter in any capacity at this point.

Comment (tareqlol) [reply packages/cna-template/template/components/Meta/Meta.tsx:69]: Fair point.

**Review by tareqlol — APPROVED** [review]

**Pull request merged**